monkey/monkey_island/cc/services/edge/edge.py

```diff
@@ -18,6 +21,7 @@ def get_all_edges() -> List[EdgeService]:
 
     @staticmethod
     def get_or_create_edge(src_node_id, dst_node_id, src_label, dst_label) -> EdgeService:
+        lock.acquire()
```

Comment:
It's safer to use a context manager

```python
with lock:
```

Then you don't have to release it (happens automatically) and you don't have to worry about what happens if you forget to release it. In this case, if line 27 raises any exception besides `DoesNotExist`, the lock will never get released and all future calls to `get_or_create_edge()` will block until the program exits.